crates/bevy_math/src/direction.rs

```diff
@@ -38,6 +38,30 @@ impl std::fmt::Display for InvalidDirectionError {
     }
 }
 
+/// Checks that a vector with the given squared length is normalized.
+///
+/// Warns for small error with a length threshold of approximately `1e-4`,
+/// and panics for large error with a length threshold of approximately `1e-2`.
+///
+/// The format used for the logged warning is `"Warning: {warning} The length is {length}`,
+/// and similarly for the error.
+#[cfg(debug_assertions)]
+fn assert_is_normalized(message: &str, length_squared: f32) {
```

Comment:
This feels like it could be a macro, but that might be overkill. It could be nice to have different messages for the warning and error though?

Reply:
Yeah, let's avoid adding macros unless we really need to.